src/object.js

```diff
@@ -163,22 +158,44 @@ module.exports = (common) => {
           })
         })
 
-        it('DAGNode with some DAGLinks', (done) => {
-          const dNode1 = new DAGNode(new Buffer('Some data 1'))
-          const dNode2 = new DAGNode(new Buffer('Some data 2'))
-          dNode1.addNodeLink('some-link', dNode2, (err) => {
-            expect(err).to.not.exist
-            ipfs.object.put(dNode1, (err, node) => {
-              expect(err).to.not.exist
-              expect(dNode1.data).to.deep.equal(node.data)
-              expect(
-                dNode1.links.map((l) => l.toJSON())
-              ).to.deep.equal(
-                node.links.map((l) => l.toJSON())
-              )
-              done()
-            })
-          })
+        it('DAGNode with a link', (done) => {
+          let node1a
+          let node1b
+          let node2
+          series([
+            (cb) => {
+              DAGNode.create(new Buffer('Some data 1'), (err, node) => {
+                expect(err).to.not.exist
+                node1a = node
+                cb()
+              })
+            },
+            (cb) => {
+              DAGNode.create(new Buffer('Some data 2'), (err, node) => {
+                expect(err).to.not.exist
+                node2 = node
+                cb()
+              })
+            },
```

Comment:
I find it super hard to read :/ as it is spread on so many lines that I have difficulty figuring out what is actually happening